### PR TITLE
efi: drop executable-stack bit from .elf file

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/boot/efi/meson.build [hunk 1]
```diff
@@ -257,6 +257,7 @@ efi_ldflags = [
         '-Wl,--warn-common',
         '-Wl,-Bsymbolic',
         '-z', 'nocombreloc',
+        '-Wl,-z,noexecstack',
```
Comment (ldv-alt) [inline src/boot/efi/meson.build]: Maybe `'-z', 'noexecstack'`, while being the exact equivalent of `'-Wl,-z,noexecstack'`, would look nicer.

**Comment by medhefgo** [comment]
This stuff is going away, why are we still bothering with it?

**Comment by keszybz** [comment]
I pushed an update with `-Wl,-z` → `-z` and with a proper commit message now. I know that this stuff is supposed to go away, but we want to keep things working without warnings until that happens.

**Review by medhefgo — APPROVED** [review]

**Review by bluca — APPROVED** [review]

**Comment by keszybz** [comment]
CentOS CI (CentOS Stream 8): TEST-75-RESOLVED, looks unrelated.

**Comment by keszybz** [comment]
jammy-amd64 and jammy-s390x are stuck, but the rpm builds on all architectures succeeded, so this should be fine.

**Pull request merged**